.NET/Microsoft.Recognizers.Text.DateTime.Tests/English/TestDateExtractor.cs

```diff
@@ -1,97 +1,142 @@
-            BasicTest("I'll go back the next day", 13, 12);
-            BasicTest("I'll go back next day", 13, 8);
-            BasicTest("I'll go back this Friday", 13, 11);
-            BasicTest("I'll go back next Sunday", 13, 11);
-            BasicTest("I'll go back last Sunday", 13, 11);
-            BasicTest("I'll go back last day", 13, 8);
-            BasicTest("I'll go back the last day", 13, 12);
-            BasicTest("I'll go back the day", 13, 7);
-            BasicTest("I'll go back this week Friday", 13, 16);
-            BasicTest("I'll go back next week Sunday", 13, 16);
-            BasicTest("I'll go back last week Sunday", 13, 16);
-            BasicTest("I'll go back 15 June 2016", 13, 12);
-            BasicTest("a baseball on may the eleventh", 14, 16);
-
-            BasicTest("I'll go back the first friday of july", 13, 24);
-            BasicTest("I'll go back the first friday in this month", 13, 30);
-
-            BasicTest("I'll go back two weeks from now", 13, 18);
-
-            BasicTest("I'll go back next week on Friday", 13, 19);
-            BasicTest("I'll go back on Friday next week", 13, 19);
-
-            BasicTest("past Monday", 0, 11);
-        }
-
-        [TestMethod]
-        public void TestDateExtractAgoLater()
-        {
-            BasicTest("I went back two months ago", 12, 14);
-            BasicTest("I'll go back two days later", 13, 14);
-            BasicTest("who did i email a month ago", 16, 11);
-        }
-    }
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using DateObject = System.DateTime;
+
+namespace Microsoft.Recognizers.Text.DateTime.English.Tests
+{
+
+    [TestClass]
+    public class TestDateExtractor
+    {
+        private readonly BaseDateExtractor extractor = new BaseDateExtractor(new EnglishDateExtractorConfiguration());
+
+        public void BasicTest(string text, int start, int length)
+        {
+            var results = extractor.Extract(text);
+            Assert.AreEqual(1, results.Count);
+            Assert.AreEqual(start, results[0].Start);
+            Assert.AreEqual(length, results[0].Length);
+            Assert.AreEqual(Constants.SYS_DATETIME_DATE, results[0].Type);
+        }
+
+        // use to generate the test cases sentences inside TestDateExtractWeekDayAndDayOfMonth function
+        // return a day of current week which the parameter refer to
+        public string GenWeekDaynDayMonthTest(int dayOfMonth)
+        {
+            var weekDay = "None";
+            if (dayOfMonth >= 1 && dayOfMonth <= 31)
+            {
+                var referenceTime = DateObject.Now;
+                var date = new DateObject(referenceTime.Year, referenceTime.Month, dayOfMonth);
+                weekDay = date.DayOfWeek.ToString();
+            }
+
+            var sentence = "I went back " + weekDay;
+            return sentence;
+        }
+
+        [TestMethod]
+        public void TestDateExtract()

```

Comment:
Where are the TestMethods to cover cases like "Thursday the 21st"?
Please create separate TestMethods for these cases, don't add test to pre-existing TestMethods.

Reply:
They are in the TestDateExtractWeekDayAndDayOfMonth. It requires the WeekOfDay to be the same with the date mentioned. Otherwise, it should be two separate time mentions. Therefore, we use a function to generate the WeekOfDay when given a date.

Reply:
Got it. Next time, please add an example in a code comment in the method. Just to make it easier to track what covers what.